bin/xcm-tools/src/cli.rs

```diff
@@ -34,3 +40,31 @@ pub struct AssetIdCmd {
     #[clap(default_value = "340282366920938463463374607431768211455")]
     pub asset_id: u128,
 }
+
+/// Helper that prints AccountId32 hash value for the derived multilocation.
+#[derive(Debug, clap::Parser)]
+pub struct Account32HashCmd {
+    /// Parachain id in case sender is from a sibling parachain.
+    #[clap(short, long, default_value = None)]
+    pub parachain_id: Option<u32>,
+    /// AccountId32 (SS58 scheme, public key) of the sender account.
+    #[clap(short, long, value_parser = account_id_32_parser)]
+    pub account_id_32: [u8; 32],
+}
+
+/// Used to parse AccountId32 as [u8; 32] from the received string.
+fn account_id_32_parser(account_str: &str) -> Result<[u8; 32], String> {
+    let re = Regex::new(r"^0x([0-9a-fA-F]{64})$").unwrap();
+    if !re.is_match(account_str) {
+        return Err(
+            "Invalid AccountId32 received. Expected format is '0x1234...4321' (64 hex digits)."
+                .into(),
+        );
+    }
+
+    let hex_acc = re.captures(account_str).unwrap().get(1).unwrap().as_str();
```

Comment:
Probably `expect` too.

Reply:
I'd like to avoid having two `expect`s since it will just reduce readability, imo.

I can add a short comment above stating that regex has already been matched and capture group 1 exists.

Reply:
Same as in previous comment.